sdk/src/main/java/com/uid2/sdk/network/DefaultNetworkSession.kt

```diff
@@ -0,0 +1,75 @@
+package com.uid2.sdk.network
+
+import com.uid2.sdk.extensions.decodeJsonToMap
+import java.net.HttpURLConnection
+import java.net.URL
+import java.net.URLEncoder
+
+/**
+ * A default implementation of NetworkSession that leverages HttpUrlConnection to make the necessary
+ * GET and POST requests.
+ */
+open class DefaultNetworkSession : NetworkSession {
+
+    override fun loadData(url: URL, request: NetworkRequest): Map<String, Any> {
+        val connection = openConnection(url).apply {
+            requestMethod = request.type.toRequestMethod()
+
+            setRequestProperty("Accept", "application/json")
+
+            // If we've been given any request headers, add them to the request.
+            request.headers.forEach {
+                addRequestProperty(it.key, it.value)
+            }
+
+            // If data was provided, this will require us to write it to the output stream.
+            request.data?.let { data ->
+                doOutput = true
+
+                outputStream.use { outputStream ->
+                    outputStream.write(data.toPostData().toByteArray(Charsets.UTF_8))
+                }
+            }
+        }
+
+        // A successful response code should be in the [200-299] range.
+        val responseCode = connection.responseCode
+        if (responseCode !in SUCCESS_CODE_RANGE_MIN..SUCCESS_CODE_RANGE_MAX) {
```

Comment:
I'm not sure we'll want to support 200 to 299 codes as the docs are pretty specific about status codes they use.  Thoughts?

https://github.com/IABTechLab/uid2docs/blob/main/api/v2/endpoints/post-token-refresh.md#response-status-codes